nomad/state/events.go

```diff
@@ -358,6 +378,24 @@ func eventFromChange(change memdb.Change) (structs.Event, bool) {
 				Service: after,
 			},
 		}, true
+	case TableHostVolumes:
+		after, ok := change.After.(*structs.HostVolume)
+		if !ok {
+			return structs.Event{}, false
+		}
+		return structs.Event{
+			Topic: structs.TopicHostVolume,
+			Key:   after.ID,
+			FilterKeys: []string{
```

Comment:
DRY bikeshed: what if `structs.HostVolume` had an `Event()` method that returned this of itself?  then it could be used in both places.

```go
// nomad/structs/host_volumes.go
func (hv *HostVolume) Event() Event {
	if hv == nil {
		return Event{}
	}
	return Event{
		Topic:     TopicHostVolume,
		Key:       hv.ID,
		Namespace: hv.Namespace,
		FilterKeys: []string{
			hv.ID,
			hv.Name,
			hv.PluginID,
		},
		Payload: &HostVolumeEvent{
			Volume: hv,
		},
	}
}
```

extra bonus bikeshed: if all the other types did this too, then this `eventFromChange` method could be real small, akin to

```go
type eventer interface {
	Event() structs.Event
}
if thing, ok := change.After.(eventer); ok {
	return thing.Event(), true
} else {
	return structs.Event{}, false
}
```

so adding something to the event stream could be (almost) as simple as satisfying that interface.

Reply:
I was thinking something along those lines while implementing this (and https://github.com/hashicorp/nomad/pull/24724). It'd improve the type safety too. But seeing as how I've already tried to shrink this PR by pulling out the CSI work to its own PR, any such refactoring should probably also be its own PR.

There's also a minor annoyance here too that the Eventer interface must be exported from `nomad/structs` none of that code will be covered by `nomad/structs` tests. The "test coverage goons" will frown, but there's really no logic there.